samples/kotlin-android-app-destinations/src/main/java/com/segment/analytics/destinations/plugins/AppsflyerDestination.kt

```diff
@@ -0,0 +1,244 @@
+package com.segment.analytics.destinations.plugins
+
+import android.app.Activity
+import android.content.Context
+import com.segment.analytics.kotlin.core.*
+import com.segment.analytics.kotlin.core.platform.DestinationPlugin
+import com.segment.analytics.kotlin.core.platform.Plugin
+import kotlinx.serialization.Serializable
+import com.appsflyer.AppsFlyerLib
+import com.appsflyer.AppsFlyerConversionListener
+import android.content.SharedPreferences
+import android.os.Bundle
+import com.appsflyer.AFInAppEventParameterName
+import com.segment.analytics.kotlin.android.plugins.AndroidLifecycle
+import com.segment.analytics.kotlin.core.platform.plugins.LogType
+import com.segment.analytics.kotlin.core.platform.plugins.log
+import com.appsflyer.deeplink.DeepLinkListener
+import com.segment.analytics.kotlin.core.utilities.getString
+import com.segment.analytics.kotlin.core.utilities.mapTransform
+import com.segment.analytics.kotlin.core.utilities.toContent
+import kotlinx.serialization.json.*
+
+/*
+This is an example of the AppsFlyer device-mode destination plugin that can be integrated with
+Segment analytics.
+Note: This plugin is NOT SUPPORTED by Segment.  It is here merely as an example,
+and for your convenience should you find it useful.
+To use it in your codebase, we suggest copying this file over and include the following
+dependencies in your `build.gradle` file:
+```
+dependencies {
+    ...
+    implementation 'com.appsflyer:af-android-sdk:6.3.2'
+    implementation 'com.android.installreferrer:installreferrer:2.2'
+}
+```
+MIT License
+Copyright (c) 2021 Segment
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*/
+
+@Serializable
+data class AppsFlyerSettings(
+    var appsFlyerDevKey: String,
+    var trackAttributionData: Boolean = false
+)
+
+class AppsFlyerDestination(
+    private val applicationContext: Context,
+    private var isDebug: Boolean = false
```

Comment:
maybe we can check if `analytics` has `logger` plugin to determine `isDebug`?

Reply:
the logger plugin is added to analytics by default, so this would result in it being always true. 

I think providing the user the option to specify when they want `debug` to be `true/false` directly is better than to derive that value via some other property. I could be convinced otherwise, but i would rather keep this

Reply:
👍 yeah. that makes sense. I also felt that's not a reliable way.